lib/logstash/inputs/lumberjack.rb

```diff
@@ -31,10 +31,13 @@ class LogStash::Inputs::Lumberjack < LogStash::Inputs::Base
 
   # This setting no longer has any effect and will be removed in a future release.
   config :max_clients, :validate => :number, :deprecated => "This setting no longer has any effect. See https://github.com/logstash-plugins/logstash-input-lumberjack/pull/12 for the history of this change"
+  
+  # The number of seconds before we raise a timeout,
+  # this option is useful to control how much time to wait if something is blocking the pipeline.
+  config :timeout, :validate => :number, :default => 5
```

Comment:
Can we call this something more specific? Not sure what to call it. "congestion_delay" maybe? Something that indicates that it's specific to an internal detection of a the pipeline being stuck.